designer/server/src/routes/forms/create.js

```diff
@@ -230,31 +222,40 @@ export default [
     async handler(request, h) {
       const { auth, payload, yar } = request
       const author = getAuthor(auth.credentials)
+      const token = auth.credentials.token
 
       // Update form metadata
-      const metadata = yar.set(sessionNames.create, {
+      yar.set(sessionNames.create, {
         ...yar.get(sessionNames.create),
         teamName: payload.teamName,
         teamEmail: payload.teamEmail
       })
 
       try {
-        const slug = slugify(metadata.title)
-
         // Create the form
-        await forms.create(payload, author)
+        const result = await forms.create(payload, author, token)
 
         // Clear form metadata
         yar.clear(sessionNames.create)
 
         /**
          * Redirect POST to GET without resubmit on back button
          */
-        return h.redirect(`/library/${slug}`).code(303)
-      } catch (cause) {
+        return h.redirect(`/library/${result?.slug}`).code(303)
+      } catch (err) {
+        if (Boom.isBoom(err) && err.data?.error === 'FormAlreadyExistsError') {
+          yar.flash('validationFailure', {
+            formErrors: create.titleFormErrors,
+            formValues: payload
+          })
+
+          // Redirect POST to GET without resubmit on back button
+          return h.redirect('/create/title').code(303).takeover()
```

Comment:
Any chance we can consolidate all these duplicated lines?

They're almost the same in both handlers

```suggestion
          await failActionDuplicate(/* params */)
```

Reply:
Resolved with 6537d6bba7ba49636d7218495cb52587df4ced18

(@colinrotherham please feel free to check/improve on the types)